cfgov/jinja2/v1/_includes/organisms/video-player.html

```diff
@@ -1,23 +1,132 @@
 {# ==========================================================================
 
-   Video Player
+   video_player.render()
 
    ==========================================================================
 
    Description:
 
-   Create a video player for the Wagtail Video Player module when given:
+   Builds Video markup when given:
 
-   value: Object defined from a StreamField block.
+   video_id:        YouTube video ID, e.g. "dQw4w9WgXcQ".
 
-   value.video_url: A string containing the YouTube embed URL to use.
+   button_pos:      String used to denote the position of the play button.
+                    Possible values: 'center (default), bottomRight'
+
+   thumbnail_url:   Optional alternate image URL to show before video is played
+                    or after it is finished.
+
+   is_fcm:          Boolean. True if this video is being rendered as part of a
```

Comment:
@anselmbradford @Scotchester is there some better way to do this, so that the VP just renders properly depending on whether or not it's contained within an FCM? If not, is there some more appropriate name for this setting, like `display_at_fixed_size`?

Reply:
Yes I would agree this has a code smell. Without digging into it I can't fully answer your question, but `display_at_fixed_size` sounds good. If you haven't already I would render the video player outside of a FCM, but with this flag set and see what happens. It should be able to be rendered with this set but outside of a FCM. If it doesn't then there's an incorrect tight coupling happening here.

Reply:
So currently (slightly modified by this branch), [the video player template](https://github.com/cfpb/cfgov-refresh/blob/fd74edddc78452b1f161ac4abfbe25ef6dbb2667/cfgov/jinja2/v1/_includes/macros/video-player.html#L52) adds `o-featured-content-module_visual` to `<div class="o-video-player">` if this boolean is true. That FCM style ([from the DS](https://github.com/cfpb/design-system/blob/1cf848f8df72f56252bbb5d99a2117f26238816d/packages/cfpb-layout/src/organisms/featured-content-module.less#L36)) then sets `position: absolute; top: 0; right: 0` to the top-level organism, which causes it to be rendered in the upper-right of the page.

So it seems like the FCM relies on the VP template having a certain class to make it render on the right side, vs the FCM itself applying that class. Is there some way of pulling that out?

Reply:
@Scotchester given how unwieldy this PR is already, and some of the questions you raise, I'm inclined to punt on doing this refactoring right now. Perhaps we can revisit with @anselmbradford after this gets merged in.